### DISCUSSION TITLE
Cookie value is automatically reset

### DISCUSSION THREAD

**Discussion by elmeerr** [remix-run/remix#9124] [Q&A] [answered]
Using cookies in a simple example just to hold some data across different part of my application and everything works locally but after deploy the values are reset:

`const myCookie = createCookie("__myCookie");`

then in my action:
```
export const action = async ({ request }: ActionFunctionArgs) => {
    const formData = await request.formData()
    const cookie = await myCookie.parse(request)
    cookie.type = formData.get('type')

    return redirect('/type', {
        headers: {
            "Set-Cookie": await myCookie.serialize(cookie),
        }
    })
}
```
this works perfectly locally and I can use my cookie wherever I want but when I deploy (Google Cloud Run) the value is set but then it goes back to. `e30%3D` (`e30=` decoded)

Any idea about what I might be doing wrong?

Thanks

**Comment by kiliman** [comment]
Here's an example showing the Remix Cookie API

⚡️ StackBlitz https://stackblitz.com/edit/remix-run-remix-xegfoa

**Comment by elmeerr** [comment, marked as answer]
@kiliman thanks for your time. The configuration is ok...the problem ended up to be on my host: GCP + Firebase Hosting does not allow any other cookie except [`__session`](https://firebase.google.com/docs/hosting/manage-cache#using_cookies) 😩
